lib/ts/recipe/session/cookieAndHeaders.ts

```diff
@@ -30,16 +30,21 @@ const frontTokenHeaderKey = "front-token";
  * @description clears all the auth cookies from the response
  */
 export function clearSession(config: TypeNormalisedInput, req: BaseRequest, res: BaseResponse, userContext: any) {
-    const transferMethod = config.getTokenTransferMethod({ req, userContext });
+    // If we can tell it's a cookie based session we are not clearing using headers
+    let transferMethod = config.getTokenTransferMethod({ req, userContext });
+
+    if (transferMethod === "MISSING_AUTH_HEADER") {
+        transferMethod = "header";
+    }
 
     const tokenTypes: TokenType[] = ["access", "refresh"];
     for (const token of tokenTypes) {
-        setToken(config, req, res, token, "", 0, userContext, transferMethod);
+        setToken(config, res, token, "", 0, transferMethod);
 
         // This is to ensure we clear the cookies as well if the user has migrated to headers,
         // because this can't be done on the client side
         if (transferMethod === "header" && isTokenInCookies(req, token)) {
-            setToken(config, req, res, token, "", 0, userContext, "cookie");
+            setToken(config, res, token, "", 0, "cookie");
         }
```

Comment:
Can we not mix migration related stuff with normal code flow? I would ideally like migration stuff to be very clearly isolated in code with a very clear code path of:
if (is using older session) {
  do migration stuff
}

Let's discuss this on call